nativeunwind/elfunwindinfo/elfehframetable.go

```diff
@@ -0,0 +1,142 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package elfunwindinfo // import "go.opentelemetry.io/ebpf-profiler/nativeunwind/elfunwindinfo"
+
+import (
+	"debug/elf"
+	"errors"
+	"fmt"
+	"sort"
+	"unsafe"
+
+	lru "github.com/elastic/go-freelru"
+	"go.opentelemetry.io/ebpf-profiler/libpf"
+	"go.opentelemetry.io/ebpf-profiler/libpf/pfelf"
+)
+
+type FDE struct {
+	PCBegin uintptr
+	PCRange uintptr
+}
+
+type EhFrameTable struct {
+	r             reader
+	hdr           *ehFrameHdr
+	fdeCount      uintptr
+	tableStartPos uintptr
+	ehFrameSec    *elfRegion
+	efm           elf.Machine
+	cieCache      *lru.LRU[uint64, *cieInfo]
+}
+
+func NewEhFrameTable(ef *pfelf.File) (EhFrameTable, error) {
```

Comment:
Maybe a personal preference, but I think it would be safer to return `nil` in error cases rather than an uninitialized `EhFrameTable` struct.
```suggestion
func NewEhFrameTable(ef *pfelf.File) (*EhFrameTable, error) {
```